# Assignment 2

In [3]:
import os
import sys
import numpy as np
import pylab as pl
import json
import os

P_0 = 0.521 and P_1 is 0.533


In [4]:
alpha=0.05
P_0=52.1*0.01
P_1=53.3*0.01 
n_0=409
n_1=564

Nt_0=P_0*n_0
Nt_1=P_1*n_1

In [5]:
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)
print sp

0.527955806783


In [6]:
sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )


sp_stdev_2y=sp_stdev((Nt_0+Nt_1)/(n_0+n_1),[n_0,n_1])
print P_0, n_0, n_1, sp_stdev_2y

0.521 409 564 0.0324224126866


In [7]:
zscore = lambda p0, p1, s : (p0-p1)/s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print z_2y

0.370114343926


In [8]:
p_2y= 1-0.6368
alpha = 0.5
def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

    
report_result(p_2y,alpha)

is the p value 0.36 smaller than the critical value 0.50? 
YES!
the Null hypothesis is rejected


In [9]:
Ntot = 973
expected = 564*409*131.579*841.421
sample_values = [[14.7*5.64,85.3*5.64],[11.9*4.09,88.1*4.09]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chisqstat(Ntot,  sample_values, expected)

1.58939333193


1.589 lies in range greater than 10 percent therefore we reject the null hypothisis.